File: java/factorial.java
public class factorial{

    static int factorial(int n){
     if(n==0)
        return 1;

        int smallAns = factorial(n-1);
        
        int ans=n*smallAns;
        return ans;
     
    }

    public static void main(String[] args) {
        System.out.println(factorial(4));
    }
}

File: c/factorial.c
#include<stdio.h>

int Factorial(int n);
int main(){
    int num;
    printf("enter the number to get the factorial:");
    scanf("%d",&num);
    printf("the factorial is %d",Factorial(num));
    return 0;

}

int factorial(int n){
    if(n==1){
        return 1;
    } else{
        return n*factorial(n-1);
    }
}
